### PR TITLE
[AI][cloudflare-prometheus-exporter] Achieve 100% test coverage

### PR DESCRIPTION
**Pull request by P6rguVyrst**
## Context

Add comprehensive unit tests for previously untested modules:

- test_cli.py: 21 tests covering CLI commands, flags, and error handling
- test_logging.py: 13 tests for CustomJsonFormatter including timestamp, logger name, and level handling
- test_cloudflare_exporter_class.py: Tests for CloudflareExporter class methods and job execution
- test_cloudflare_exporter_parsers.py: Tests for all parser functions including edge cases and multiple zone scenarios
- test_error_scenarios.py: Error handling and API error responses
- test_run_exporter.py: run_exporter function with various configurations

Test fixtures follow best practices:
- All fixtures centralized in conftest.py
- JSON test data stored in tests/data/ directory
- YAML config files in tests/data/config/
- No embedded JSON/dicts in Python test files

Coverage improvements:
- cli.py: 0% -> 100%
- logging.py: 0% -> 100%
- cloudflare_exporter.py: 98% -> 100%
- Overall: 80% -> 100%

All tests follow TDD principles and test meaningful behavior, not just lines for coverage metrics.

```
================================================================================================== tests coverage ===================================================================================================
_________________________________________________________________________________ coverage: platform darwin, python 3.9.21-final-0 __________________________________________________________________________________

Name                                         Stmts   Miss  Cover
----------------------------------------------------------------
cloudflare_exporter/__init__.py                  5      0   100%
cloudflare_exporter/cli.py                      24      0   100%
cloudflare_exporter/cloudflare_exporter.py     112      0   100%
cloudflare_exporter/gql/__init__.py             14      0   100%
cloudflare_exporter/logging.py                  12      0   100%
cloudflare_exporter/metrics.py                  20      0   100%
----------------------------------------------------------------
TOTAL                                          187      0   100%
========================================================================================== 94 passed, 2 warnings in 0.51s ===========================================================================================
```

## Checklist
- [x] Change meets or does not compromise the [Baseline Security Requirements](https://transferwise.atlassian.net/wiki/spaces/EKB/pages/434929973/Baseline+Security+Requirements)

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1,26 +1,56 @@
...
+    """Zone object with multiple keys (warning case)."""
+    with open("tests/data/zones/multiple_keys_zone.json") as data:
+        return json.load(data)
+
+
+@pytest.fixture(scope="session")
+def sample_config():
+    """Sample configuration for CLI testing."""
+    with open("tests/data/config/valid_config.yaml") as data:
+        return yaml.safe_load(data)
```
Comment (P6rguVyrst) [inline tests/conftest.py]: Haven't validated the validity of those json responses, but this can be done when making changes to functionality and working with actual API repsonses.

**Review by P6rguVyrst — COMMENTED** [review]
Looks fine. Tested the change to cloudflare_exporter.py to make sure it didn't break. Used my personal domain to test nothing broke.

The test coverage is more here to enable testing for intent and rewriting it in another language if desired.

No functional changes-- just increasing unit test coverage.
Real review is more required when changing the functional codebase.

**Review by P6rguVyrst — COMMENTED** [review]
Authors can't approve their PRs.

**Review by andrew-marwood — APPROVED** [review]
